AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin.md

```diff
@@ -0,0 +1,93 @@
+- The object is analyzed according to the model schema, association data (post.id and "post" field name) is stored at "comments" by creating and storing `AppSyncModelMetadata` at the "comments" key.
+- The object is then serialized and then decoded to the post instance. The `Post` decoder will instantiate the fields as normally while the "comments" field will be decoded by the `List<Comment>` that delegates its logic to the `AppSyncModelListDecoder`. 
+- `AppSyncListDecoder` checks that the data can be successfully decoded to an `AppSyncModelMetadata` and stores this information in an `AppSyncListProvider` when instantiating a "not loaded" list with association data.
+The developer has the post instance and can lazy load the comments upon accessing it.
+```swift
+foreach comment in post.comments {
+   /// comments is loaded before the first element is returned.
+}
+```
+- The comments are implicitly loaded upon access by taking the association data and performing a query to get the list of comments by its post id. Storing the result in the list object, and returning that to the caller that is performing the access.
+The implicit load is the same as performing an explicit call to retrieve a list of models, given a limit and condition.
+```swift
+Amplify.API.query(.paginatedList(Comment.self, limit: 100, where: Comment.keys.post == post.id))
+```
+
+4. As a developer, I can customize my request to retrieve multiple levels of data at once
+```swift
+let document = """
+  query getBlog($id: ID!) {
+    getBlog(id: $id) {
+      id
+      post {
+        items {
+          id
+          comments {
+            items {
+              id
+            }
+            nextToken
+          }
+        }
+        nextToken
+      }
+    }
+  }
+  """
+```
+- `GraphQLResponseDecoder` will skip argumenting the response data at "post" and "comments" with association data since the response data already contains the payload to be decoded to the list.
+- `AppSyncListDecoder` checks that the data can be successfully decoded to `AppSyncListResponse` and instantaites a loaded list of post and list of comments respectively in the chain of decoders. 
+
+This advanced use case is not without caveats on integrity of the subsequent APIs calls performed from the list object. For example, `hasNextPage` relies on the selection set to contain `nextToken`, so if this is excluded from the selection set, then `hasNextPage` will always return false. `getNextPage` does not retrieve the next page according to the associated parent since association data was never added to the list provider. By providing this successful decoding flow `AppSyncListResponse`, the data represents more of a snapshot, and assumes that the developer understand what they are trying to achieve with the customization. Alternatively, developers can go to the full extent of modifying the response type as well to `AppSyncListResponse` to decode exactly what the AppSync service returns in the response.
```

Comment:
```suggestion
This is an advanced use case, and has some caveats regarding the subsequent API calls performed from the `List`:
- `hasNextPage()` relies on the selection set to contain `nextToken`, so if this is excluded from the selection set, then `hasNextPage()` will always return `false`.
- `getNextPage(completion:)` does not retrieve the next page according to the associated parent since association data was never added to the `List` provider.

In this flow the returned data represents more of a snapshot, and assumes that the developer understands what they are trying to achieve with the customization.

Alternately, developers can go to the full extent of modifying the response type as well to `AppSyncListResponse` to control exactly what the AppSync service returns in the response.
```